src/Symfony/Component/DependencyInjection/ConfigurationResolverTrait.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+namespace Symfony\Component\DependencyInjection;
+
+use Symfony\Component\ExpressionLanguage\Expression;
+
+trait ConfigurationResolverTrait
+{
+    protected function resolve($value)
+    {
+        if (is_string($value) && strlen($value) > 0 && $value[0] == '@') {
```

Comment:
Instead of `strlen() > 0` you should use `'' !== $value`, also strict check for this: `'@' === $value[0]`.